spec/services/search_query_parser_spec.rb

```diff
@@ -0,0 +1,45 @@
+RSpec.describe SearchQueryParser do
```

Comment:
Spec looks great but its only testing the 'happy' path at the moment. It could do with tests for behaviour under failure conditions as well, particularly for things like api requests where network issues are out of your control.

There's some testing around failure conditions in the frontend's `ApiEntity` spec

https://github.com/trade-tariff/trade-tariff-frontend/blob/main/spec/lib/api_entity_spec.rb#L100

There's also more info on the WebMock documentation - see https://github.com/bblimke/webmock

Reply:
I've added an unhappy path: when the query param is 0 it return 400 (and an error is raised by Faraday).